staging/src/k8s.io/apiserver/pkg/endpoints/openapi/openapi.go

```diff
@@ -189,3 +198,12 @@ func (d *DefinitionNamer) GetDefinitionName(name string) (string, spec.Extension
 	}
 	return friendlyName(name), nil
 }
+
+func (d *DefinitionNamer) GetDefinitionNamev3(name string) (string, spec.Extensions) {
```

Comment:
can you give some examples in a go doc? How do we make sure the result is unique? Do we have to?

Reply:
New comments are added. The result must be unique. But there is a small problem with the implementation:

There are two kinds of return values:

1. GVK in form of GROUP.VERSION.KIND
2. Type package path with `/` replaced with `.`. Like `io.k8s.api.core.v1.EndpointSubset`. The reason this is needed is because `EndpointSubset` does not have a GVK associated with it. 

All GVKs cannot conflict with one another.
All types paths cannot conflict with one another.

It is very unlikely for there to be a collision, but it is possible. I can not think of a way to uniqueify the two classes of return values other than maybe replacing `.` with `-` so that for example `io.k8s.api.core.v1.EndpointSubset` becomes `io-k8s-api-core-v1-EndpointSubset`. That way all values of this form will be unique from any dot-separated GVK. But this is ugly. Any thoughts @sttts?

Reply:
Am still unsure whether we want these package paths like `io.k8s.api.core.v1.EndpointSubset`. Those are types used somewhere inside a GVK, right? This is still some kind of leaking of implementation.

Also note that `io.k8s.api.core.v1.EndpointSubset` does not really version that type. It could theoretically be that another type uses `io.k8s.api.core.v1.EndpointSubset` in another version, at least through aggregation. There we actually deepequal compare them and assign a new, unique name per unique schema.

Am undecided. Maybe leave it like it is for the moment, have no better idea.